ui/src/components/ConfigurationForm/ConfigurationForm.component.tsx

```diff
@@ -23,19 +24,64 @@ import {
   getJSONImplementationOptions,
   mapEffectTypeToJSONImplementation,
 } from './ConfigurationForm.helpers';
-import { useNavigate } from 'react-router-dom';
+import { useSearchParams, useNavigate } from 'react-router-dom';
 import { ApiCallAddons } from '../ApiCallAddons';
 import { ConfigurationDataContext, resetFormData, setFormData } from '../../contexts';
+import { parse, stringifyUrl } from 'query-string';
 
 interface ConfigurationFormProps {
   isEmbedded?: boolean;
 }
 
+interface SnackbarConfig {
+  open: boolean;
+  severity?: AlertColor;
+  message?: String;
+}
+
 export const ConfigurationForm: React.FC<ConfigurationFormProps> = ({ isEmbedded = false }) => {
   const navigate = useNavigate();
   const [{ formData }, contextDispatch] = useContext(ConfigurationDataContext);
   const { call, clearError, isLoading, errorMessage } = useApiCall();
   const { classes, cx } = useStyles({ isEmbedded });
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  useEffect(() => {
+    if (searchParams.toString() !== '') {
+      const qp = parse(searchParams.toString(), { parseBooleans: true });
+      starterValidationSchema
+        .isValid(qp)
+        .then(isValid => {
+          if (isValid) {
+            const queryParams = qp as StarterRequest;
+            let key: keyof StarterRequest;
+            for (key in queryParams) {
+              form.setValue(key, queryParams[key]);
+            }
+            setSnackbar({
+              open: true,
+              severity: 'info',
+              message: 'Linked configuration was applied.',
+            });
+          } else {
+            setSnackbar({
+              open: true,
+              severity: 'warning',
+              message: 'Linked configuration is not valid therefore it was not applied.',
+            });
+          }
+        })
+        .catch(_ => {
+          setSnackbar({
+            open: true,
+            severity: 'warning',
+            message: 'Validation of linked configuration failed therefore it was not applied.',
+          });
+        });
+      setSearchParams({});
+    }
+  });
```

Comment:
Hooks usually have dependency arrays, to run effects only when elements of those dependency arrays change (ref: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects). In the past I've encountered subtle bugs when I left it off. But I'm not sure how important it is here, as you have if statements that prevents doing too much work here. What do you think @kkolasinski, should dependency array be filled here?

Reply:
there was a dependency array filled with `setSearchParams` but `yarn lint` complained with the following
```
83:6  warning  React Hook useEffect has missing dependencies: 'form' and 'searchParams'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```
so I have removed it altogether... having said that I'm opened for suggestions ;)

Reply:
AFAIK empty deps array causes the `useEffect` hook to be ran once.

From what I understand, the idea is to set the form values by getting them from the URL.

If we're not planning on updating the URL in real time, we should be fine with omitting the dependency array - the URL params are read once during the initial render, so we don't need to track any changes.

Reply:
That is the intention: perform the operation in question once when the form is displayed for the first time.